src/HIE/Bios/Cradle.hs

```diff
@@ -637,33 +637,51 @@ readProcessWithOutputFile
   -> FilePath -- ^ Process to call.
   -> [String] -- ^ Arguments to the process.
   -> IO (ExitCode, [String], [String], [String])
-readProcessWithOutputFile l ghcProc work_dir fp args =
-  withSystemTempFile "bios-output" $ \output_file h -> do
-    hSetBuffering h LineBuffering
-    old_env <- getEnvironment
+readProcessWithOutputFile l ghcProc work_dir fp args = do
+  old_env <- getEnvironment
+
+  withHieBiosOutput old_env $ \output_file -> do
     let (ghcPath, ghcArgs) = case ghcProc of
             Just (p, a) -> (p, unwords a)
             Nothing ->
               ( fromMaybe "ghc" (lookup hieBiosGhc old_env)
               , fromMaybe "" (lookup hieBiosGhcArgs old_env)
               )
+
     -- Pipe stdout directly into the logger
     let process = (readProcessInDirectory work_dir fp args)
                       { env = Just
                               $ (hieBiosGhc, ghcPath)
                               : (hieBiosGhcArgs, ghcArgs)
-                              : ("HIE_BIOS_OUTPUT", output_file)
+                              : (hieBiosOutput, output_file)
                               : old_env
                       }
-        -- Windows line endings are not converted so you have to filter out `'r` characters
-        loggingConduit = (C.decodeUtf8  C..| C.lines C..| C.filterE (/= '\r')  C..| C.map T.unpack C..| C.iterM l C..| C.sinkList)
+
+    -- Windows line endings are not converted so you have to filter out `'r` characters
+    let  loggingConduit = (C.decodeUtf8  C..| C.lines C..| C.filterE (/= '\r')  C..| C.map T.unpack C..| C.iterM l C..| C.sinkList)
     (ex, stdo, stde) <- sourceProcessWithStreams process mempty loggingConduit loggingConduit
-    !res <- force <$> hGetContents h
+    existsFile <- doesFileExist output_file
+    res <- if existsFile
+             then withFile output_file ReadMode $ \handle -> do
+                    hSetBuffering handle LineBuffering
+                    !res <- force <$> hGetContents handle
+                    return res
+             else return ""
```

Comment:
This should be an error rather than returning empty string?

Reply:
well, it seems there is test cases that actually dont write anything in the HIE_BIOS_OUTPUT (they failed before ad2a675 cause they dont even created the file to write in) so i assumed that it is a known-allowed case (see https://travis-ci.org/github/mpickering/hie-bios/jobs/683972142#L764 f.e.)
maybe is it the other way around and tests were wrong?

Reply:
well it turns out that `readProcessWithOutputFile` is used to make calls (well only one of them actually does it) that are not related with the args and consume the stdout/stderr:

https://github.com/mpickering/hie-bios/blob/f0abff9c855ea7e6624617df669825f3f62f723b/src/HIE/Bios/Cradle.hs#L489-L490

so it seems it is fine in some cases (one?) to ignore the fact that the call has not written anything to HIE_BIOS_OUTPUT

Reply:
I would really rather not return empty string here as imagine this situation.

* User consults the cradle, the file doesn't exist for some unknown reason
* The cradle reports "" as the output
* ghcide starts a session with no options
* User sees error about import of module failing

Now to get from the first step to the last step is not trivial to realise unless you are familar with how all the parts work.

Reply:
ok, i've removed the file existence as you suggested check so it will crash early